File: src/features/users/dto/create-user.dto.ts
import { IsArray, IsBoolean, IsEmail, IsEnum, IsOptional, IsString, Matches } from 'class-validator';
import { STRONG_PASSWORD_MESSAGE, STRONG_PASSWORD_REGEX } from '../../auth/dto/password-rules';
import { Role } from '../../../common/roles';

export class CreateUserDto {
  @IsString()
  username: string;

  @IsOptional()
  @IsString()
  firstName?: string;

  @IsOptional()
  @IsString()
  lastName?: string;

  @IsEmail()
  email: string;

  @IsString()
  @Matches(STRONG_PASSWORD_REGEX, { message: STRONG_PASSWORD_MESSAGE })
  password: string;

  @IsOptional()
  @IsEnum(Role)
  role?: Role;

  @IsOptional()
  @IsArray()
  @IsEnum(Role, { each: true })
  roles?: Role[];

  @IsOptional()
  @IsString()
  orgId?: string;

  @IsOptional()
  @IsString()
  verificationTokenHash?: string | null;

  @IsOptional()
  verificationTokenExpires?: Date | null;

  @IsOptional()
  @IsString()
  resetTokenHash?: string | null;

  @IsOptional()
  resetTokenExpires?: Date | null;

  @IsOptional()
  isEmailVerified?: boolean;

  @IsOptional()
  @IsBoolean()
  isOrgOwner?: boolean;
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/users/CreateUserDto.java
package com.mytegroup.api.dto.users;

import com.mytegroup.api.common.enums.Role;
import com.mytegroup.api.dto.auth.PasswordRules;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateUserDto {
    
    @NotBlank(message = "Username is required")
    private String username;
    
    private String firstName;
    
    private String lastName;
    
    @NotBlank(message = "Email is required")
    @Email(message = "Email must be valid")
    private String email;
    
    @NotBlank(message = "Password is required")
    @Pattern(regexp = PasswordRules.STRONG_PASSWORD_REGEX, message = PasswordRules.STRONG_PASSWORD_MESSAGE)
    private String password;
    
    private Role role;
    
    private List<Role> roles;
    
    private String orgId;
    
    private String verificationTokenHash;
    
    private LocalDateTime verificationTokenExpires;
    
    private String resetTokenHash;
    
    private LocalDateTime resetTokenExpires;
    
    private Boolean isEmailVerified;
    
    private Boolean isOrgOwner;
    
    public void setEmail(String email) {
        this.email = email != null ? email.toLowerCase().trim() : null;
    }
    
    public void setUsername(String username) {
        this.username = username != null ? username.trim() : null;
    }
    
    public void setFirstName(String firstName) {
        this.firstName = firstName != null ? firstName.trim() : null;
    }
    
    public void setLastName(String lastName) {
        this.lastName = lastName != null ? lastName.trim() : null;
    }
}
